### PR TITLE
Mod/exercise6

### PR DESCRIPTION
**Pull request by JoseLavariega**
Main modifications:
- New task definitions for Separating Axis Theorem Exercises
- Data generation for SAT exercises 
- Modification of visualization code to take in optional arguments, if students choose to visualize the separating axis that they create. 
- Additional checker function to compare two lines, for polygon projection tasks. 
- Changes structure of ex06 (after running the checker) to include the new exercises. 
- Docstrings on all functions in collision_primitives and collision_checker, even legacy functions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
 # PDM4AR-exercises
+## New Branch
```
Comment (alezana) [inline README.md:2]: could you remove these lines? we don't want them in master

**Review by alezana — CHANGES_REQUESTED** [review]
@JoseLavariega @ysli16 I think you need to update the exercise instructions.
I.e., check the .md file in the docs folder

**Comment by alezana** [comment]
@JoseLavariega @ysli16 still missing the docs update

**Comment by JoseLavariega** [comment]
Docs markdown has been updated

**Pull request merged**